tests/test_json_schema.py

```diff
@@ -2692,73 +2683,202 @@ class Model(BaseModel):
         pet: Union[Cat, Dog, Lizard] = Field(..., discriminator='pet_type')
 
     assert Model.model_json_schema() == {
```

Comment:
@Julian Example 1:

Reply:
I'm staring strictly at the JSON Schemas here not so much the implementation code, so if anything I say is ugly to actually implement that's perfectly believable! And indeed I'm no expert (yet?) on OpenAPI even so hope I don't say anything too ridiculous.